File: python/resistor-color-trio/resistor_color_trio.py
color_list = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']
value_list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

def color_code(color):
    for index, c in enumerate(color_list):
        if color == c:
            return value_list[index]
        
def value(colors):
    band_one, band_two, multiplier, *_ = colors
    return (color_code(band_one) * 10 + color_code(band_two)) * 10**color_code(multiplier) # type: ignore

def label(colors):
    value_length = len(str(value(colors)))
    if value_length == 4 or value_length == 5 or value_length == 6:
        return str(value(colors))[:-3] + " kiloohms"
    elif value_length == 7 or value_length == 8 or value_length == 9:
        return str(value(colors))[:-6] + " megaohms"
    elif value_length == 10 or value_length == 11 or value_length == 12:
        return str(value(colors))[:-9] + " gigaohms"
    else:
        return str(value(colors)) + " ohms"

File: java/resistor-color-trio/src/main/java/ResistorColorTrio.java
import java.util.Arrays;

class ResistorColorTrio {
    String label(String[] colors) {
        long colorValue = (long) (value(colors) * Math.pow(10, colorCode(colors[2])));
        int length = String.valueOf(colorValue).length();
        if (length >= 4 && length <= 6) {
            return String.valueOf(colorValue / 1000) + " kiloohms";
        } else if (length >= 7 && length <= 9) {
            return String.valueOf(colorValue / 1_000_000) + " megaohms";
        } else if (length >= 10 && length <= 12) {
            return String.valueOf(colorValue / 1_000_000_000) + " gigaohms";
        }
        return String.valueOf(colorValue) + " ohms";
    }

    private final int COLOR_QUANTITY = 2;

    int value(String[] colors) {
        int sum = 0;
        int exponent = 0;
        for (int i = COLOR_QUANTITY - 1; i >= 0; i--) {
            sum += colorCode(colors[i]) * Math.pow(10, exponent);
            exponent++;
        }
        return sum;
    }

    int colorCode(String color) {
        return Arrays.asList(colors()).indexOf(color);
    }

    String[] colors() {
        return new String[] {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet",
                "grey", "white"};
    }
}
